src/live_restore/live_restore_fs.c

```diff
@@ -502,81 +427,39 @@ __live_restore_fh_lock(WT_FILE_HANDLE *fh, WT_SESSION *wt_session, bool lock)
 }
 
 /*
- * __live_restore_remove_extlist_hole --
+ * __live_restore_fh_fill_bit_range --
  *     Track that we wrote something by removing its hole from the extent list. Callers must hold
  *     the extent list write lock.
  */
-static int
-__live_restore_remove_extlist_hole(
+static void
+__live_restore_fh_fill_bit_range(
   WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh, WT_SESSION_IMPL *session, wt_off_t offset, size_t len)
 {
-    WTI_LIVE_RESTORE_HOLE_NODE *hole, *tmp, *new, *prev_hole;
-    wt_off_t write_end;
-
-    WT_ASSERT_ALWAYS(session, __wt_rwlock_islocked(session, &lr_fh->ext_lock),
+    WT_ASSERT_ALWAYS(session, __wt_rwlock_islocked(session, &lr_fh->bitmap_lock),
       "Live restore lock not taken when needed");
-    __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE, "REMOVE HOLE %s: %" PRId64 "-%" PRId64,
-      lr_fh->iface.name, offset, WTI_OFFSET_END(offset, len));
-
-    write_end = WTI_OFFSET_END(offset, len);
-    hole = lr_fh->destination.hole_list_head;
-    prev_hole = NULL;
-    while (hole != NULL) {
-        if (write_end < hole->off)
-            /* We won't find any more overlapping holes. Stop searching. */
-            break;
-
-        if (offset <= hole->off && write_end >= WTI_EXTENT_END(hole)) {
-            /* The write fully overlaps a hole. Delete it. */
-            __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE,
-              "Fully overlaps hole %" PRId64 "-%" PRId64, hole->off, WTI_EXTENT_END(hole));
-
-            tmp = hole;
-            if (prev_hole == NULL)
-                lr_fh->destination.hole_list_head = hole->next;
-            else
-                prev_hole->next = hole->next;
-            hole = hole->next;
-            __wt_free(session, tmp);
-            continue;
-        } else if (offset > hole->off && write_end < WTI_EXTENT_END(hole)) {
-            /* The write is entirely within the hole. Split the hole in two. */
-
-            __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE,
-              "Fully contained by hole %" PRId64 "-%" PRId64, hole->off, WTI_EXTENT_END(hole));
 
-            /* First create the hole to the right of the write. */
-            WT_RET(__live_restore_alloc_extent(session, write_end + 1,
-              (size_t)(WTI_EXTENT_END(hole) - write_end), hole->next, &new));
+    /* If the file is complete or the write falls outside the bitmap return. */
```

Comment:
Nit: "...outside the bitmap, return" Add a comma.

Reply:
ba9df39c3